rest-api/data_gen/fake_participant_generator.py

```diff
@@ -295,38 +343,150 @@ def _make_measurement_resource(self, measurement, qualifier_set):
         } for qualifier in qualifiers]
     return resource
 
-  def _make_measurement_entry(self, measurement, time_str, participant_id, qualifier_set):
-    resource = self._make_measurement_resource(measurement, qualifier_set)
+  def _get_measurement_values(self, measurement_resources):
+    return [resource['valueQuantity']['value'] for resource in measurement_resources]
+
+  def _get_related(self, measurement_resources):
+    related_list = []
+    related_urls = set()
+    for measurement_resource in measurement_resources:
+      resource_related = measurement_resource.get('related')
+      if resource_related:
+        for related in resource_related:
+          # Make sure we don't repeat references to the same qualifier.
+          url = related['target']['reference']
+          if url not in related_urls:
+            related_list.append(related)
+            related_urls.add(url)
+    return related_list
+
+  def _mean(self, values):
+    return round(sum(values) / len(values), 1)
+
+  def _find_closest_two_mean(self, measurement_resources):
+    measurement_values = self._get_measurement_values(measurement_resources)
+    # Find the mean of the two closest values; or if they are equally distant from each other,
+    # find the mean of all three.
+    delta_0_1 = abs(measurement_values[0] - measurement_values[1])
+    delta_0_2 = abs(measurement_values[0] - measurement_values[2])
+    delta_1_2 = abs(measurement_values[1] - measurement_values[2])
+    if delta_0_1 < delta_0_2:
+      if delta_0_1 < delta_1_2:
+        return self._mean(measurement_values[0:2])
+      else:
+        return self._mean(measurement_values[1:])
+    elif delta_0_1 == delta_0_2:
+      return self._mean(measurement_values)
+    else:
+      if delta_0_2 < delta_1_2:
+        return self._mean([measurement_values[0], measurement_values[2]])
+      else:
+        return self._mean(measurement_values[1:])
+
+  def _calculate_last_two_mean(self, measurement_resources):
+    measurement_values = self._get_measurement_values(measurement_resources)
+    return self._mean(measurement_values[1:])
+
+  def _find_components(self, measurement_resources, submeasurement):
```

Comment:
Might be clearer to call this `_find_components_by_code`, and pass in just the `code` instead of a whole submeasurement. (Or better, `_find_components_by_coding`, to match systems too.)